src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Abstractions/IGeneratedSymbolMacro.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Core.Contracts;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Abstractions
+{
+    /// <summary>
+    /// An interface for macro created via generated symbols.
+    /// </summary>
+    public interface IGeneratedSymbolMacro : IMacro
+    {
+        /// <summary>
+        /// Evaluates macro defined via generated symbol (<see cref="IGeneratedSymbolConfig"/>).
+        /// The result of macro evaluation is modification of variable collection <paramref name="variables"/>.
```

Comment:
Btw. do we ever want to modify any existing variables, or only add?
It would likely require more involved change in contracts... so not worth it. I was just wondering if we foresee any such scenario where altering existing variables is wanted

Reply:
Now it is assumed that you can also modify existing variables, i.e. `IVariableCollection` is kind of in-out.